hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/TestRMWebServicesForCSWithPartitions.java

```diff
@@ -106,76 +105,79 @@ protected void configureServlets() {
       bind(ResourceManager.class).toInstance(rm);
       serve("/*").with(GuiceContainer.class);
     }
-  });
+  };
 
-  public class GuiceServletConfig extends GuiceServletContextListener {
-    @Override
-    protected Injector getInjector() {
-      return injector;
-    }
+  static {
+    GuiceServletConfig.setInjector(
+        Guice.createInjector(new WebServletModule()));
   }
 
   private static void setupQueueConfiguration(
-      CapacitySchedulerConfiguration conf, ResourceManager rm) {
+      CapacitySchedulerConfiguration config, ResourceManager resourceManager) {
```

Comment:
Thank you for updating the pull request! Mostly looks good to me.
(nit) Unused argument `resourceManager` can be removed. I'm +1 if that is addressed.